src/network/core/game_info.cpp

```diff
@@ -198,6 +200,21 @@ void SerializeNetworkGameInfo(Packet *p, const NetworkServerGameInfo *info)
 	/* Update the documentation in game_info.h on changes
 	 * to the NetworkGameInfo wire-protocol! */
 
+	/* NETWORK_GAME_INFO_VERSION = 5 */
+	{
+		/* Only send GameScript details if server is running some Gamescript */
+		GameInfo* gsinfo = Game::GetInfo();
+		if (gsinfo == nullptr) {
+			p->Send_bool(false); //has gamescript?
+		}
```

Comment:
Is there a difference? Both will send zero byte and bool(false) or uint8(0) seems more understandable to me.

Reply:
Not sure we are speaking the same language here :)

My suggestion is to not send this bool at all. Just run: Send_string(name); Send_uint32(version);. And if those are "" and -1, you know there was not GS loaded. That is a lot simpler throughout the code, as it saves the constant checking if the bool was set yes/no :)

PS: did you know that for every comment you make on GitHub, I get a lovely email :D Not a real problem, but just so you know. So answer like "OK" and "will be fixed" .. I will see that next push, don't worry about it :)

Reply:
It save one check when reading received packet. A check for existing GS still has to be there, so you don't read properties of null and also check for GUI has to stay, it will just change to something like 

if(gs_version != -1) DrawString()...

I would consider having extra bool variable better for understanding code than arbitrary -1 value, but both are sufficient, so I guess, why not.